src/python/pants/engine/selectors.py

```diff
@@ -16,13 +16,36 @@ class Get:
   """Experimental synchronous generator API.
 
   May be called equivalently as either:
-    # verbose form: Get(product_type, subject_declared_type, subject)
-    # shorthand form: Get(product_type, subject_type(subject))
+    # verbose form: Get(product, subject_declared_type, subject)
+    # shorthand form: Get(product, subject_declared_type(<constructor args for subject>))
   """
   product: Type
   subject_declared_type: Type
   subject: Any
 
+  def __await__(self) -> Generator[Any, Any, Any]:
```

Comment:
Can any of these Generator arguments be `None`? I genuinely don't know - the `typing.Generator` still confuses me.

Reply:
This generator does all three of `yield`ing a value, receiving a value from that `yield`, then `return`ing it, which means that a type exists for all three of `[YieldType, SendType, ReturnType]`, so `Any` is the lowest-effort solution that appears to work here.

Reply:
Got it. Thanks!